ClasscastException in kubernetes-maven-plugin

### Describe the bug

I run into the following exception with the maven-kubernetes-plugin:

```
> mvn -e clean install
...
rg.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.11.0:build (docker-build-app) on project kubernetes-sample: A type incompatibility occurred while executing org.eclipse.jkube:kubernetes-maven
-plugin:1.11.0:build: class java.util.HashMap cannot be cast to class java.lang.String (java.util.HashMap and java.lang.String are in module java.base of loader 'bootstrap')
...
Caused by: java.lang.ClassCastException: class java.util.HashMap cannot be cast to class java.lang.String (java.util.HashMap and java.lang.String are in module java.base of loader 'bootstrap')
    at org.eclipse.jkube.kit.common.util.MavenConfigurationExtractor.addAsList (MavenConfigurationExtractor.java:73)
    at org.eclipse.jkube.kit.common.util.MavenConfigurationExtractor.getElement (MavenConfigurationExtractor.java:54)
    at org.eclipse.jkube.kit.common.util.MavenConfigurationExtractor.getElement (MavenConfigurationExtractor.java:59)
    at org.eclipse.jkube.kit.common.util.MavenConfigurationExtractor.getElement (MavenConfigurationExtractor.java:59)
    at org.eclipse.jkube.kit.common.util.MavenConfigurationExtractor.getElement (MavenConfigurationExtractor.java:59)
    at org.eclipse.jkube.kit.common.util.MavenConfigurationExtractor.extract (MavenConfigurationExtractor.java:41)
    at org.eclipse.jkube.kit.common.util.MavenUtil.getPlugins (MavenUtil.java:73)
    at org.eclipse.jkube.kit.common.util.MavenUtil.convertMavenProjectToJKubeProject (MavenUtil.java:205)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.doExecute (AbstractDockerMojo.java:434)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.execute (AbstractDockerMojo.java:413)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)

```
Maven-Version:
```
> mvn -v
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: C:\ProgrammeDE\Java\tools\apache-maven-3.8.6
Java version: 17.0.4, vendor: Eclipse Adoptium, runtime: C:\ProgrammeDE\Java\jdk17.0.4_8-win64
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

```


### Eclipse JKube version

1.11.0

### Component

Kubernetes Maven Plugin

### Apache Maven version

other (please specify in additional context)

### Gradle version

None

### Steps to reproduce

1. Add the following maven plugin to pom.xml that has also the kubernetes-maven-plugin configured
```xml
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-checkstyle-plugin</artifactId>
        <version>3.2.1</version>
	<configuration>
		<checkstyleRules>
			<module name="Checker">
				<module name="TreeWalker">
					<module name="IllegalImport">
						<property name="severity" value="warning" />
						<property name="illegalPkgs" value="sun, edu"/>
					</module>

					<module name="RedundantImport"/>
				</module>
			</module>
		</checkstyleRules>
		....
	</configuration>
</plugin>
```
2. Run `mvn install`

The `<module name="RedundantImport"/>`  causes the exception. When I change it to 
```
<module name="RedundantImport">
     <property name="severity" value="warning" />
</module>
```
then the exception does no longer occur.


### Expected behavior

The configuration of the checkstyle-maven-plugin is a valid Maven configuration.
The kubernetes-maven-plugin should work with such a configuration

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.25.3

### Environment

Windows

### Eclipse JKube Logs

_No response_

### Sample Reproducer Project

_No response_

### Additional context

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,7 @@ Usage:
 ### 1.12-SNAPSHOT
 * Fix #1273: Deprecate `jkube.io` annotation prefix in favor of `jkube.eclipse.org` for JKubeAnnotations
 * Fix #2079: Bump kubernetes-client to 6.5.0
+* Fix #2093: ClassCastException when extracting plugins from pom
 
 ### 1.11.0 (2023-02-16)
 * Fix #1316: Add support for adding InitContainers via plugin configuration

--- a/jkube-kit/common-maven/src/main/java/org/eclipse/jkube/kit/common/util/MavenConfigurationExtractor.java
+++ b/jkube-kit/common-maven/src/main/java/org/eclipse/jkube/kit/common/util/MavenConfigurationExtractor.java
@@ -68,7 +68,7 @@ public class MavenConfigurationExtractor {
         final Object insertedValue = conf.get(currentElement.getName());
         if (insertedValue instanceof List) {
             ((List) insertedValue).add(currentElement.getValue());
-        } else {
+        } else if (insertedValue instanceof String) {
             final List<String> list = new ArrayList<>();
             list.add((String) insertedValue);
             list.add(currentElement.getValue());

--- a/jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenConfigurationExtractorTest.java
+++ b/jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenConfigurationExtractorTest.java
@@ -135,6 +135,27 @@ class MavenConfigurationExtractorTest {
                 .containsEntry("a", expected);
     }
 
+    @Test
+    void should_parse_list_with_inner_elements() {
+
+        // Given
+        final Plugin fakePlugin = createFakePlugin("<a>" +
+                "<b><c>c1</c><c>c2</c></b>" +
+                "<b/>" +
+                "</a>");
+
+        // When
+        final Map<String, Object> config = MavenConfigurationExtractor.extract((Xpp3Dom) fakePlugin.getConfiguration());
+
+        // Then
+        final Map<String, Object> expectedC = new HashMap<>();
+        expectedC.put("c", Arrays.asList("c1", "c2"));
+        final Map<String, Object> expected = new HashMap<>();
+        expected.put("b", expectedC);
+
+        assertThat(config).containsEntry("a", expected);
+    }
+
     private Plugin createFakePlugin(String config) {
         Plugin plugin = new Plugin();
         plugin.setArtifactId("jkube-maven-plugin");

